### PR TITLE
Support using an http.cookiefile secret to clone

### PR DESCRIPTION
**Pull request by fejta**
Enables prow to use google cloud source repositories and other https
based systems, with something like the following:

```bash
bazel run //prow/cmd/clonerefs -- \
  --repo my-project,my-repo=master \
  --uri-prefix 'https://source.developers.google.com/p/{{.Org}}/r/{{.Repo}}' \
  --cookiefile ~/my/cookies
```

Will result in clonerefs doing the equivalent of:
```bash
  git -c http.cookiefile=~/my/cookies fetch https://source.developers.google.com/p/my-project/r/my-repo
```

ref #9270 

/assign @stevekuznetsov @mithrav @krzyzacy @BenTheElder

### CODE REVIEW

File: prow/apis/prowjobs/v1/types.go [hunk 1]
```diff
@@ -319,7 +322,7 @@ func (r Refs) String() string {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// VirtualMachineList is a list of VirtualMachine resources
+// ProwJobList is a list of ProwJob resources
```
Comment (BenTheElder) [inline prow/apis/prowjobs/v1/types.go]: !? now I want to look at the history for this line :joy:

Comment (krzyzacy) [reply prow/apis/prowjobs/v1/types.go]: :man_facepalming:

**Comment by BenTheElder** [comment]
Generally LGTM.
Do the podutils currently support overriding these settings per-job or per-repo? We might want to consider that in the future so it's possible to use this without supplying the cookiefile to _every_ job on the same Prow.

**Comment by fejta** [comment]
> Do the podutils currently support overriding these settings per-job or per-repo

The config can have specify a default decoration config (which [we do](https://github.com/kubernetes/test-infra/blob/a202e595a33ac92ab503f913f2d710efabd3de21/prow/config.yaml#L5))

Each prowjob can also define any overrides in the spec ([src](https://github.com/kubernetes/test-infra/blob/a202e595a33ac92ab503f913f2d710efabd3de21/prow/apis/prowjobs/v1/types.go#L140))

**Comment by fejta** [comment]
> how would I supply the cookie file on a prowjob?

```yaml


# Specify it on the job
presubmits:
  org/repo:
  - name: my-job-with-manual-overrides
     decoration_config:
       ssh_key_secrets: ["old-ssh-secret"]
       cookiefile_secret: new-cookiefile-secret

# Or for all jobs managed by plank
plank:
  default_decoration_config:
    ssh_key_secrets: ["we-dont-set-this-currently"]
    cookiefile_secret: we-could-set-this-in-the-future
```

**Comment by fejta** [comment]
Correct I'm going to write a controller which updates the secret every hour or so

**Comment by fejta** [comment]
(not in this PR though)

**Comment by krzyzacy** [comment]
ok sgtm
/lgtm
/hold

**Pull request merged**